recipes/whereami/all/conanfile.py

```diff
@@ -0,0 +1,95 @@
+from conans import ConanFile, CMake, tools
+import os
+import textwrap
+
+required_conan_version = ">=1.33.0"
+
+
+class WhereamiConan(ConanFile):
+    name = "whereami"
+    description = "Locate the current executable and the current module/library on the file system"
+    topics = ("conan", "whereami", "introspection", "getmodulefilename",
+              "dladdr", "executable-path", "getexecutablepath")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/gpakosz/whereami"
+    license = "MIT"  # + WTFPLv2
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        url = self.conan_data["sources"][self.version]["url"]
+        commit = url[url.rfind("/")+1:url.find(".tar.gz")]
+        extracted_folder = self.name + "-" + commit
+        os.rename(extracted_folder, self._source_subfolder)
```

Comment:
```suggestion
        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
```

Have you tried it? strip_root helps a lot.